locals.tf

```diff
@@ -34,12 +29,47 @@ locals {
           ]
         }]
       }
+      metrics = {
+        serviceMonitor = {
+          enabled = var.enable_service_monitor
+        }
+      }
       rootUser     = "root"
       rootPassword = random_password.minio_root_secretkey.result
-      # call created config_minio variable to use it the chart 
-      users    = var.config_minio.users
-      buckets  = var.config_minio.buckets
-      policies = var.config_minio.policies
-    }
+      users        = var.config_minio.users
+      buckets      = var.config_minio.buckets
+      policies     = var.config_minio.policies
+
+      }, var.oidc.issuer_url != "" && var.oidc.client_id != "" && var.oidc.client_secret != "" ? merge({
```

Comment:
I'm happy to receive alternatives on this line if you think there is a better/cleaner way.